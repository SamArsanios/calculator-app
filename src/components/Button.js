import React from 'react';
import PropTypes from 'prop-types';

export default function Button(props) {
  const { name, clickHandler } = props;
  const handleClick = (button) => clickHandler(button);
  return (
    <button
      className="button"
      type="button"
      onClick={handleClick.bind(this, name)}
    >
      {name}
    </button>
  );
}

Button.propTypes = {
  name: PropTypes.string,
  clickHandler: PropTypes.func.isRequired,
};

Button.defaultProps = {
  name: '',
};
