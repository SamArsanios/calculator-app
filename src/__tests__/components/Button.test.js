import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Button from '../../components/Button';

const test = () => { };

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Button name="1" clickHandler={test} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with provided name', () => {
    render(<Button name="button1" clickHandler={test} />);
    const button = screen.getByRole('button');
    expect(button).toHaveTextContent('button1');
  });
});
