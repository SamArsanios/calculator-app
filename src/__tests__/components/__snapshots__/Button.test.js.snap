// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="button orange "
  onClick={[Function]}
  type="button"
>
  1
</button>
`;
